daml/Marketplace/Transfer.daml

```diff
@@ -53,3 +53,53 @@ template DepositTransferRequest
 
       DepositTransferRequest_Reject : ()
         do return ()
+
+template DepositDebitRequest
+  with
+    owner : Party
+    ownerAccount : Account
+    depositCid : ContractId AssetDeposit
+  where
+    let senderSig = head $ toList ownerAccount.id.signatories
+        custodian = senderSig
+    signatory owner
+
+    controller custodian can
+      DepositDebitRequest_Approve : Asset
```

Comment:
Might be good to include Reject choices for completeness

Reply:
👍  done